MongoDB default function deleted clang warning

**Describe the bug**
Compile warning

**To Reproduce**
Seen on CI macOS builds

**Expected behavior**
no warning

**Logs**
```
In file included from src/MongoDBTest.cpp:18:
/Users/runner/work/poco/poco/MongoDB/include/Poco/MongoDB/PoolableConnectionFactory.h:125:20: warning: explicitly defaulted move assignment operator is implicitly deleted [-Wdefaulted-function-deleted]
        PooledConnection& operator=(PooledConnection&&) = default;
                          ^
/Users/runner/work/poco/poco/MongoDB/include/Poco/MongoDB/PoolableConnectionFactory.h:128:49: note: move assignment operator of 'PooledConnection' is implicitly deleted because field '_pool' is of reference type 'Poco::ObjectPool<Connection, Connection::Ptr> &' (aka 'ObjectPool<Poco::MongoDB::Connection, SharedPtr<Poco::MongoDB::Connection>> &')
        Poco::ObjectPool<Connection, Connection::Ptr>& _pool;
                                                     ^
```

**Please add relevant environment information:**
 - macOS 
 - clang

--- a/MongoDB/include/Poco/MongoDB/PoolableConnectionFactory.h
+++ b/MongoDB/include/Poco/MongoDB/PoolableConnectionFactory.h
@@ -89,11 +89,13 @@ namespace MongoDB {
 
 class PooledConnection
 	/// Helper class for borrowing and returning a connection automatically from a pool.
+	/// Note that the connection pool is not expected to be deleted during the lifetime
+	/// of an instance of PooledConnection.
 {
 public:
-	PooledConnection(Poco::ObjectPool<Connection, Connection::Ptr>& pool) : _pool(pool)
+	PooledConnection(Poco::ObjectPool<Connection, Connection::Ptr>& pool) : _pool(&pool)
 	{
-		_connection = _pool.borrowObject();
+		_connection = _pool->borrowObject();
 	}
 
 	virtual ~PooledConnection()
@@ -102,7 +104,7 @@ public:
 		{
 			if (_connection)
 			{
-				_pool.returnObject(_connection);
+				_pool->returnObject(_connection);
 			}
 		}
 		catch (...)
@@ -125,7 +127,7 @@ public:
 	PooledConnection& operator=(PooledConnection&&) = default;
 
 private:
-	Poco::ObjectPool<Connection, Connection::Ptr>& _pool;
+	Poco::ObjectPool<Connection, Connection::Ptr>* _pool;
 	Connection::Ptr _connection;
 };
 

